.registry/resources/allocatedcluster.ui-schema.yaml

```diff
@@ -0,0 +1,83 @@
+initialConfig: true
+configSections:
+- title: CNP Kubernetes Cluster
+  description: Provide the following details to provision a Kubernetes cluster
+  items:
+  - name: region
+    controlType: singleSelect
+    path: .spec.region
+    title: Region
+    description: Region of the resources that will be deployed.
+    enum:
+    - 'ap-east-1'
+    - 'ap-northeast-1'
+    - 'ap-northeast-2'
+    - 'ap-south-1'
+    - 'ap-southeast-1'
+    - 'ap-southeast-2'
+    - 'ca-central-1'
+    - 'eu-central-1'
+    - 'eu-north-1'
+    - 'eu-west-1'
+    - 'eu-west-2'
+    - 'eu-west-3'
+    - 'me-south-1'
+    - 'sa-east-1'
+    - 'us-east-1'
+    - 'us-east-2'
+    - 'us-west-2'
+    validation:
+      - required: true
+- title: Generate Provider Secret
+  description: You can use the following command to generate an AWS Provider secret that contains your default profile credentials. If you'd like to use a different account profile, you can specify it by changing the value of AWS_PROFILE environment variable in the beginning of the command.
+  items:
+  - name: additionalInfoSnippet
+    controlType: codeSnippet
+    codeText: AWS_PROFILE=default && echo -e "[default]\naws_access_key_id = $(aws configure get aws_access_key_id --profile $AWS_PROFILE)\naws_secret_access_key = $(aws configure get aws_secret_access_key --profile $AWS_PROFILE)"
+- title: Enter Your Generated Provider Secret
+  description: Once you have successfully generated the provider secret using the command above you can now enter it below.
+  items:
+  - name: providerSecret
+    controlType: secretInput
+    path:
+    title: Provider Secret
+    rows: 4
+    secret:
+      namePath: .spec.credentialsSecretRef.name
+      namespacePath: .spec.credentialsSecretRef.namespace
+      keyPath: .spec.credentialsSecretRef.key
+      keyValue: credentialsKeyBody
+    validation:
+    - required: true
+- title: Cluster Settings
+  description: Provide the following cluster configuration parameters
+  items:
+  - name: nodeCount
+    controlType: singleInput
+    type: integer
+    path: .spec.nodes.count
+    title: Node Count
+    description: The number of cluster nodes
+    default: 1
+    validation:
+    - minimum: 1
+    - maximum: 100
+  - name: nodeSize
+    controlType: singleSelect
+    path: .spec.nodes.size
+    title: Node Size
+    description: Size of node
+    enum:
+    - 'small'
+    - 'medium'
+    - 'large'
+    validation:
+      - required: true
+  - name: splunkLoggingToken
+    controlType: singleInput
+    type: string
+    path: .spec.logging.token
+    title: Splunk API Token
+    description: API token to connect to Splunk for logging
+    validation:
+      - required: true
```

Comment:
Is it worth attempting to sketch out what this file would look like if it were supplementing the [relevant OpenAPI schema](https://github.com/negz/example-cnp/blob/fc2ab2affcce293c5cd98b153e0c061b2b8ee466/clusters.cnp.example.org/definition.yaml#L44) rather than duplicating (parts of) it?

Reply:
yes, based on previous discussion had 3 phases:
1) ui meta as currently defined
2) support OpenAPI schema as a graceful fallback
3) deprecate ui meta fields that are overlapping with OpenAPI schema

so that would be the 3rd phase